raiden/message_handler.py

```diff
@@ -240,6 +240,9 @@ def handle_message_refundtransfer(
         raiden: "RaidenService", message: RefundTransfer
     ) -> List[StateChange]:
         chain_state = views.state_from_raiden(raiden)
+        # XXX: Not sure about this one. What should we do if the LockedTransfer has no
+        #      route_states due to validation?
+        # AFAIK the routes in the received transfer aren't relevant for the refund anyway.
```

Comment:
> the routes in the received transfer aren't relevant for the refund anyway.

that's technically correct, as there is no guarantee that the reverse route has enough funding available. however, as we're doing refunds on a best effort base anyway, I think trying the reverse route is the best we can do. otherwise there needs to be careful consideration if paying for a new refund route is worth it considering the opportunity cost of the locked funds.

Reply:
Sure, but the focus of this comment was on the case where the route_states would be empty due to validation errors (which should be exceedingly rare anyway).

In the happy case the route_states will be present as normal.